packages/firestore/src/remote/bloom_filter.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// @ts-ignore
+import md5 from 'crypto-js/md5';
+
+export class BloomFilter {
+  private readonly bitSize: number;
+
+  constructor(
+    private readonly bitmap: Uint8Array,
+    padding: number,
+    private readonly hashCount: number
+  ) {
+    this.bitSize = this.bitmap.length * 8 - padding;
+  }
+
+  getBitSize(): number {
+    return this.bitSize;
+  }
+
+  mightContain(document: string): boolean {
+    // Hash the string using md5
+    const md5Hash = md5HashString(document);
+    const hash64: Uint8Array = md5HashToBytes(md5Hash.words);
+    // Interpret the hashed value as two 64-bit chunks as unsigned integers, encoded using 2’s
+    // complement using little endian.
+    const hash1 = Buffer.from(hash64).readBigUInt64LE(0);
+    const hash2 = Buffer.from(hash64).readBigUInt64LE(8);
```

Comment:
Use Buffer to read byte array into BigInt directly.
**AI:** check if we can use Buffer if the client is using browser, not node.